python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -354,15 +354,29 @@ def removeNodeAttribute(self, attrib_name):
         if attrib_name in self._data.keys():
             del self._data[attrib_name]
 
-    def removeNodeIndexCacheAttribute(self, attrib_name):
-        if self._index_cache and attrib_name in self._index_cache.keys():
-            del self._index_cache[attrib_name]
-
     def setNodeIndexCache(self, new_index_cache):
+        if self.fullyLoadedFromDisk():
+            logger.debug("Warning: Setting IndexCache data on live object, please avoid!")
         setattr(self, '_index_cache', new_index_cache)
 
-    def getNodeIndexCache(self):
+    def getNodeIndexCache(self, force_cache=False):
```

Comment:
Wouldn't it be far more productive to add them in one go, one file at a time and not end up with numerous commits and recommits fixing typos around the code